src/main/java/ch/njol/skript/sections/EffSecOpenInventory.java

```diff
@@ -0,0 +1,283 @@
+			this.inventory = inventory;
+			this.players = players;
+		}
+
+		public Inventory getInventory() {
+			return inventory;
+		}
+
+		public Player[] getPlayers() {
+			return players;
+		}
+
+		@Override
+		public HandlerList getHandlers() {
+			throw new IllegalStateException();
+		}
+
+	}
+
+	static {
+		EventValues.registerEventValue(InventorySectionEvent.class, Inventory.class, InventorySectionEvent::getInventory);
+		EventValues.registerEventValue(InventorySectionEvent.class, Player[].class, InventorySectionEvent::getPlayers);
+		Skript.registerSection(EffSecOpenInventory.class,
+				"(show|create|open) %inventory/inventorytype% (to|for) %players%",
+				"open [a[n]] " + OpenableInventorySyntax.construct() + " [view|window|inventory] (to|for) %players%",
+
+				"close [the] inventory [view] (of|for) %players%",
+				"close %players%'[s] inventory [view]");
+	}
+
+	private @Nullable OpenableInventorySyntax syntax;
+	private @Nullable Expression<?> object;
+	private @Nullable Trigger trigger;
+
+	private Expression<Player> players;
+	private boolean open;
+
+	@Override
+	@SuppressWarnings("unchecked")
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult,
+			SectionNode sectionNode, List<TriggerItem> triggerItems) {
+
+		object = exprs.length > 1 ? exprs[0] : null;
+		if (matchedPattern == 1) {
+			open = true;
+			if (!parseResult.tags.isEmpty()) { // %-inventory% was not used
+				syntax = OpenableInventorySyntax.valueOf(parseResult.tags.get(0).toUpperCase(Locale.ENGLISH));
+				if (syntax.getVersion() != null && !Skript.isRunningMinecraft(syntax.getVersion())) {
+					Skript.error("Opening an inventory of type '" + syntax.toString().toLowerCase(Locale.ENGLISH) + "' is only present on Minecraft version " + syntax.getVersion());
+					return false;
+				}
+				if (!syntax.doesMethodExist()) {
+					Skript.error(syntax.getMethodError());
+					return false;
+				}
+			}
+		}
+
+		players = (Expression<Player>) exprs[exprs.length - 1];
+		if (exprs[0] instanceof Literal<?> literal && literal.getSingle() instanceof InventoryType inventoryType && !inventoryType.isCreatable()) {
+			Skript.error("Cannot create an inventory of type " + Classes.toString(inventoryType));
+			return false;
+		}
+
+		if (open && hasSection())
+			trigger = loadCode(sectionNode, "open inventory", InventorySectionEvent.class);
+
+		return true;
+	}
+
+	@Override
+	protected TriggerItem walk(Event event) {
+		if (object != null) {
+			Inventory inventory = null;
+			Object o = object.getSingle(event);
+			if (o instanceof Inventory i) {
+				inventory = i;
+			} else if (o instanceof InventoryType inventoryType && inventoryType.isCreatable()) {
+				inventory = createInventory(inventoryType);
+			}
+			if (inventory == null)
+				return super.walk(event, false);
+
+			Player[] players = this.players.getArray(event);
+			if (players.length > 0 && trigger != null) {
```

Comment:
Due to earlier errors, the trigger will always be null since loadCode was never called.